### PR TITLE
Add tradePolicy retrieved from selected facets

### PR DESCRIPTION
**Pull request by ullayne02**
#### What problem is this solving?
Allow to Jumbo retrieve the Price from a specific trade policy 

#### How should this be manually tested?

[alssports](https://tradepolicy--alssports.myvtex.com)
[polishop](https://tradepolicy--polishop.myvtex.com/)
[dzarm](https://tradepolicy--dzarm.myvtex.com)
[jumbo - graphql](https://ullayne--jumbo.myvtex.com/_v/private/vtex.search-graphql@0.35.0/graphiql/v1)

query: 
```
query {
  productSearch(fullText: "id:43534", selectedFacets: [{key:"trade-policy" value: "30"}], from: 0, to: 39) {
    products {
      productId
      productName
      brand
      properties {
        name
        values
      }
      productReference
      categoryId
      categoryTree {
        id
        name
        children {
          id
          name
          children {
            id
            name
          }
        }
      }
      productClusters {
        id
        name
      }
      linkText
      items {
        itemId
        name
        unitMultiplier
        measurementUnit
        images {
          imageUrl
          imageTag
        }
        referenceId {
          Key
          Value
        }
        sellers {
          sellerId
          sellerName
          commertialOffer {
            Price
            ListPrice
            PriceWithoutDiscount
            AvailableQuantity
          }
        }
      }
    }
  }
}
```

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [ ] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Changed
+
+- Use trade policy from seleted facets in product search query.
```
Comment (thalytafabrine) [inline CHANGELOG.md]:
```suggestion
- Use trade policy from selected facets in `productSearch` query. 
```

File: node/clients/checkout.ts [hunk 2]
```diff
@@ -17,16 +17,16 @@ export class Checkout extends JanusClient {
...
-    const queryString = channel ? `?sc=${channel}` : ''
+    const queryString = channel ? `?sc=${tradePolicy || channel}` : ''
     return queryString
   }
 
   public simulation = (simulation: SimulationPayload) =>
     this.post<OrderForm>(
-      this.routes.simulation(this.getChannelQueryString()),
+      this.routes.simulation(this.getChannelQueryString(simulation.tradePolicy)),
       simulation,
```
Comment (thalytafabrine) [inline node/clients/checkout.ts:30]: does this route expect to receive simulation with the `tradePolicy` prop? if not, we need to remove it before sending `simulation` here

Comment (ullayne02) [reply node/clients/checkout.ts:30]: the simulation route? if so, yes! :D

Comment (thalytafabrine) [reply node/clients/checkout.ts:30]: hmm i don't think so. I believe that the `tradepolicy` should be passed on the item or by query string, can you remove this information from the `simulation` object before sending?

Comment (ullayne02) [reply node/clients/checkout.ts:30]: I'm passing as a dependency now :D

**Review by thalytafabrine — APPROVED** [review]

File: node/commons/compatibility-layer.ts [hunk 3]
```diff
@@ -208,6 +208,7 @@ const getSimulationPayloads = (product: SearchProduct) => {
         id: item.itemId,
         quantity: 1,
         seller: seller.sellerId
+
```
Comment (chrsmutti) [inline node/commons/compatibility-layer.ts]:
```suggestion
```

File: node/resolvers/search/index.ts [hunk 4]
```diff
@@ -579,7 +587,8 @@ export const queries = {
     const productResolver = args.productOriginVtex
       ? productsCatalog
       : productsBiggy
-    const convertedProducts = await productResolver({ ctx, simulationBehavior, searchResult: result })
+
+    const convertedProducts = await productResolver({ ctx, simulationBehavior, searchResult: result, tradePolicy})
```
Comment (chrsmutti) [inline node/resolvers/search/index.ts:590]:
```suggestion
    const convertedProducts = await productResolver({ ctx, simulationBehavior, searchResult: result, tradePolicy })
```

File: node/commons/products.ts [hunk 5]
```diff
@@ -6,16 +6,17 @@ interface ConvertProductInput {
...
-export const productsBiggy = async ({ searchResult, ctx, simulationBehavior = 'default' }: ConvertProductInput) => {
+export const productsBiggy = async ({ searchResult, ctx, simulationBehavior = 'default', tradePolicy }: ConvertProductInput) => {
   const { segment } = ctx.vtex
   const checkout = ctx.clients.checkout
   const products: any[] = []
 
   searchResult.products.forEach((product: any) => {
     try {
-      products.push(convertBiggyProduct(product, checkout, simulationBehavior, segment?.channel, segment?.priceTables, segment?.regionId))
+      products.push(convertBiggyProduct(product, checkout, simulationBehavior, tradePolicy || segment?.channel, segment?.priceTables, segment?.regionId))
```
Comment (chrsmutti) [inline node/commons/products.ts]:
```suggestion
      products.push(convertBiggyProduct(product, checkout, simulationBehavior, tradePolicy ?? segment?.channel, segment?.priceTables, segment?.regionId))
```

File: node/clients/checkout.ts [hunk 6]
```diff
@@ -17,16 +17,16 @@ export class Checkout extends JanusClient {
     })
   }
 
-  private getChannelQueryString = () => {
+  private getChannelQueryString = (tradePolicy?: string) => {
     const { segment } = this.context as CustomIOContext
     const channel = segment && segment.channel
-    const queryString = channel ? `?sc=${channel}` : ''
+    const queryString = channel ? `?sc=${tradePolicy || channel}` : ''
```
Comment (chrsmutti) [inline node/clients/checkout.ts]:
```suggestion
    const queryString = channel ? `?sc=${tradePolicy ?? channel}` : ''
```

**Review by chrsmutti — CHANGES_REQUESTED** [review]

File: node/commons/products.ts (hunk 5 above)
Comment (ullayne02) [reply node/commons/products.ts]: What's the difference?

Comment (chrsmutti) [reply node/commons/products.ts]: No coercing shenanigans.
image

**Review by chrsmutti — APPROVED** [review]

**Pull request merged**